crates/llm-ls/src/retrieval.rs

```diff
@@ -359,17 +369,21 @@ impl SnippetRetriever {
         let col = self.db.get_collection("code-slices").await?;
         let mut encoding = self.tokenizer.encode(snippet.clone(), true)?;
         encoding.truncate(512, 1, TruncationDirection::Right);
+        let batch = vec![encoding];
```

Comment:
Ah yes, we're in the `search` method. We still need to handle the overflowing of the truncation I think. What would be your approach here?

Reply:
Ideally, I think that the snippet should be splitted into chunks, and added to the prompt only if they are close enough from the last chunk before the cursor. But this would also means several embedding per search, so there is a latency issue to address. What do you think ?

Reply:
I see what you mean, what do you think of getting a window that would fit the embeddings size around the cursor?
May be some extra work but surely might be what will work best

Reply:
Yes that seems like a good solution !